src/pip/_internal/operations/prepare.py

```diff
@@ -73,35 +73,49 @@ class IsWheel(DistAbstraction):
 
     def dist(self, finder):
         return list(pkg_resources.find_distributions(
-            self.req.source_dir))[0]
+            self.req_to_install.source_dir))[0]
 
-    def prep_for_dist(self):
+    def prep_for_dist(self, finder):
         # FIXME:https://github.com/pypa/pip/issues/1112
         pass
 
 
 class IsSDist(DistAbstraction):
-
     def dist(self, finder):
-        dist = self.req.get_dist()
+        dist = self.req_to_install.get_dist()
         # FIXME: shouldn't be globally added.
         if dist.has_metadata('dependency_links.txt'):
             finder.add_dependency_links(
                 dist.get_metadata_lines('dependency_links.txt')
             )
         return dist
 
-    def prep_for_dist(self):
-        self.req.run_egg_info()
-        self.req.assert_source_matches_version()
+    def prep_for_dist(self, finder):
+        # build_requirements = self.req_to_install.get_requires()
+        logger.info("Installing build dependencies")
+        # Install the build requirements
+        # with self.req.build_envirionment as prefix:
+        #    finder = copy(finder)
+        #    finder.format_control = FormatControl(set(), set())
+        #    urls = [finder.find_requirement(InstallRequirement.from_line(r),
+        #            upgrade=False).url for r in build_requirements]
+        #     #    # TODO: Use single process with recursion handling
+        #  args = [sys.executable, '-m', 'pip', 'install',
+        #           '--ignore-installed',
+        #           '--prefix', prefix] + list(urls)
+        #    with open_spinner("Installing build dependencies") as spinner:
+        #        call_subprocess(args, show_stdout=False, spinner=spinner)
+
```

Comment:
FYI this never worked but this is roughly where it should go. That's currently beyond the scope of this though.

Reply:
Uh, what??? You just said above "I don't see any barriers to merging this ASAP" and now you're saying there's something here that never worked? Please clarify.

Reply:
This in in `master` now but does not work, hence #4647. Hence this does not introduce any regressions.

Reply:
@pradyunsg I just want to coordinate this ahead of time with you. This is the code that was moved here to install the build requirements but it is commented out because it was a mess. Do you have a way to setup the environment without using a subprocess?